SUI message box windows no longer display

One or more regressions in SUI windows have been introduced, where they don't appear to the player.

Example of the newbie quest being granted when spawning:

```
29-06-24 15:06:01.095 T: [QuestService] user/Venne Uthaxars received quest quest/c_newbie_start
29-06-24 15:06:01.096 T: [QuestService] user/Venne Uthaxars activating task 0 of quest quest/c_newbie_start
29-06-24 15:06:01.110 E: Fatal Exception while processing intent: SuiWindowIntent(player=Player[226295 NAME=user / Venne Uthaxars STATE=ZONED_IN], window=com.projectswg.holocore.resources.support.global.zone.sui.SuiMessageBox@3f735f22, windowId=-1, event=NEW)
29-06-24 15:06:01.110 E: Exception in thread "intent-processor-8" java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
29-06-24 15:06:01.110 E: Caused by: java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
29-06-24 15:06:01.110 E:     me.joshlarson.jlcommon.control.Service.invoke(Service.java:135)
29-06-24 15:06:01.110 E:     me.joshlarson.jlcommon.control.Service.lambda$registerIntentHandlers$1(Service.java:119)
29-06-24 15:06:01.110 E:     me.joshlarson.jlcommon.control.IntentManager$IntentRunner.broadcast(IntentManager.java:214)
29-06-24 15:06:01.110 E:     me.joshlarson.jlcommon.control.IntentManager$IntentExecutor.run(IntentManager.java:269)
29-06-24 15:06:01.110 E:     me.joshlarson.jlcommon.utilities.ThreadUtilities.safeRun(ThreadUtilities.java:47)
29-06-24 15:06:01.110 E:     me.joshlarson.jlcommon.concurrency.ThreadPool$ThreadExecutor.threadExecutor(ThreadPool.java:174)
29-06-24 15:06:01.110 E:     java.base/java.lang.Thread.run(Thread.java:1583)
29-06-24 15:06:01.110 E:   Exception in thread "intent-processor-8" java.lang.reflect.InvocationTargetException: null
29-06-24 15:06:01.111 E:   Caused by: java.lang.reflect.InvocationTargetException: null
29-06-24 15:06:01.111 E:       java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:115)
29-06-24 15:06:01.111 E:       java.base/java.lang.reflect.Method.invoke(Method.java:580)
29-06-24 15:06:01.111 E:       me.joshlarson.jlcommon.control.Service.invoke(Service.java:133)
29-06-24 15:06:01.111 E:       me.joshlarson.jlcommon.control.Service.lambda$registerIntentHandlers$1(Service.java:119)
29-06-24 15:06:01.111 E:       me.joshlarson.jlcommon.control.IntentManager$IntentRunner.broadcast(IntentManager.java:214)
29-06-24 15:06:01.111 E:       me.joshlarson.jlcommon.control.IntentManager$IntentExecutor.run(IntentManager.java:269)
29-06-24 15:06:01.111 E:       me.joshlarson.jlcommon.utilities.ThreadUtilities.safeRun(ThreadUtilities.java:47)
29-06-24 15:06:01.111 E:       me.joshlarson.jlcommon.concurrency.ThreadPool$ThreadExecutor.threadExecutor(ThreadPool.java:174)
29-06-24 15:06:01.111 E:       java.base/java.lang.Thread.run(Thread.java:1583)
29-06-24 15:06:01.111 E:     Exception in thread "intent-processor-8" java.lang.NullPointerException: null
29-06-24 15:06:01.111 E:     Caused by: java.lang.NullPointerException: null
29-06-24 15:06:01.111 E:         com.projectswg.common.data.sui.SuiBaseWindow.getLength(SuiBaseWindow.kt:240)
29-06-24 15:06:01.111 E:         com.projectswg.common.network.packets.swg.zone.server_ui.SuiCreatePageMessage.encode(SuiCreatePageMessage.java:58)
29-06-24 15:06:01.112 E:         com.projectswg.holocore.resources.support.global.network.NetworkClient.addToOutbound(NetworkClient.kt:215)
29-06-24 15:06:01.112 E:         com.projectswg.holocore.resources.support.global.network.NetworkClient.player$lambda$2(NetworkClient.kt:77)
29-06-24 15:06:01.112 E:         com.projectswg.holocore.resources.support.global.player.Player.sendPacket(Player.java:168)
29-06-24 15:06:01.112 E:         com.projectswg.holocore.services.support.global.zone.sui.SuiService.displayWindow(SuiService.kt:106)
29-06-24 15:06:01.112 E:         com.projectswg.holocore.services.support.global.zone.sui.SuiService.handleSuiWindowIntent(SuiService.kt:56)
29-06-24 15:06:01.112 E:         java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
29-06-24 15:06:01.112 E:         java.base/java.lang.reflect.Method.invoke(Method.java:580)
29-06-24 15:06:01.112 E:         me.joshlarson.jlcommon.control.Service.invoke(Service.java:133)
29-06-24 15:06:01.112 E:         me.joshlarson.jlcommon.control.Service.lambda$registerIntentHandlers$1(Service.java:119)
29-06-24 15:06:01.112 E:         me.joshlarson.jlcommon.control.IntentManager$IntentRunner.broadcast(IntentManager.java:214)
29-06-24 15:06:01.112 E:         me.joshlarson.jlcommon.control.IntentManager$IntentExecutor.run(IntentManager.java:269)
29-06-24 15:06:01.112 E:         me.joshlarson.jlcommon.utilities.ThreadUtilities.safeRun(ThreadUtilities.java:47)
29-06-24 15:06:01.112 E:         me.joshlarson.jlcommon.concurrency.ThreadPool$ThreadExecutor.threadExecutor(ThreadPool.java:174)
29-06-24 15:06:01.112 E:         java.base/java.lang.Thread.run(Thread.java:1583)
```

--- a/src/main/java/com/projectswg/holocore/resources/support/global/zone/sui/SuiMessageBox.kt
+++ b/src/main/java/com/projectswg/holocore/resources/support/global/zone/sui/SuiMessageBox.kt
@@ -95,6 +95,10 @@ class SuiMessageBox : SuiWindow() {
 			}
 		}
 
+	init {
+		super.suiScript = "Script.messageBox"
+	}
+
 	private fun setShowRevertButton(shown: Boolean) {
 		val value = shown.toString()
 		setProperty("btnRevert", "Enabled", value)

--- a/src/main/java/com/projectswg/holocore/services/support/global/zone/LoginService.java
+++ b/src/main/java/com/projectswg/holocore/services/support/global/zone/LoginService.java
@@ -63,11 +63,7 @@ import me.joshlarson.jlcommon.control.IntentHandler;
 import me.joshlarson.jlcommon.control.Service;
 
 import java.net.SocketAddress;
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
+import java.util.*;
 import java.util.concurrent.CopyOnWriteArrayList;
 
 public class LoginService extends Service {
@@ -77,12 +73,14 @@ public class LoginService extends Service {
 	
 	private final Map<String, List<CreatureObject>> players;
 	private final PswgUserDatabase userDatabase;
-	
+	private final Collection<Galaxy> galaxies;
+
 	public LoginService() {
-		this(PswgDatabase.INSTANCE.getUsers());
+		this(Collections.singletonList(ProjectSWG.INSTANCE.getGalaxy()), PswgDatabase.INSTANCE.getUsers());
 	}
 
-	public LoginService(PswgUserDatabase pswgUserDatabase) {
+	public LoginService(Collection<Galaxy> galaxy, PswgUserDatabase pswgUserDatabase) {
+		this.galaxies = galaxy;
 		userDatabase = pswgUserDatabase;
 		this.players = Collections.synchronizedMap(new HashMap<>());
 	}
@@ -160,7 +158,7 @@ public class LoginService extends Service {
 		} else if (isPasswordValid(user, loginRequest.getPassword())) {
 			StandardLog.onPlayerEvent(this, player, "logged in from %s", loginRequest.getSocketAddress());
 			onSuccessfulLogin(user, player);
-			player.sendPacket(new HoloLoginResponsePacket(true, "", getGalaxies(), getCharacters(user.getUsername())));
+			player.sendPacket(new HoloLoginResponsePacket(true, "", galaxies, getCharacters(user.getUsername())));
 		} else {
 			StandardLog.onPlayerEvent(this, player, "failed to login [incorrect password] from %s", loginRequest.getSocketAddress());
 			onInvalidUserPass(player);
@@ -263,7 +261,7 @@ public class LoginService extends Service {
 		LoginEnumCluster cluster = new LoginEnumCluster();
 		LoginClusterStatus clusterStatus = new LoginClusterStatus();
 		List<SWGCharacter> characters = getCharacters(player.getAccountId());
-		for (Galaxy g : getGalaxies()) {
+		for (Galaxy g : galaxies) {
 			cluster.addGalaxy(g);
 			clusterStatus.addGalaxy(g);
 		}
@@ -279,13 +277,7 @@ public class LoginService extends Service {
 	private boolean isPasswordValid(UserMetadata user, String password) {
 		return userDatabase.authenticate(user, password);
 	}
-	
-	private List <Galaxy> getGalaxies() {
-		List<Galaxy> galaxies = new ArrayList<>();
-		galaxies.add(ProjectSWG.INSTANCE.getGalaxy());
-		return galaxies;
-	}
-	
+
 	private List<SWGCharacter> getCharacters(String accountId) {
 		List<SWGCharacter> characters = new ArrayList<>();
 		List<CreatureObject> creatures = this.players.get(accountId);

--- a/src/test/java/com/projectswg/holocore/test/resources/GenericPlayer.java
+++ b/src/test/java/com/projectswg/holocore/test/resources/GenericPlayer.java
@@ -1,5 +1,5 @@
 /***********************************************************************************
- * Copyright (c) 2023 /// Project SWG /// www.projectswg.com                       *
+ * Copyright (c) 2024 /// Project SWG /// www.projectswg.com                       *
  *                                                                                 *
  * ProjectSWG is the first NGE emulator for Star Wars Galaxies founded on          *
  * July 7th, 2011 after SOE announced the official shutdown of Star Wars Galaxies. *
@@ -30,6 +30,7 @@ package com.projectswg.holocore.test.resources;
 import com.projectswg.common.data.CRC;
 import com.projectswg.common.data.encodables.tangible.Posture;
 import com.projectswg.common.data.location.Location;
+import com.projectswg.common.network.NetworkProtocol;
 import com.projectswg.common.network.packets.SWGPacket;
 import com.projectswg.common.network.packets.swg.zone.*;
 import com.projectswg.common.network.packets.swg.zone.baselines.Baseline;
@@ -82,6 +83,7 @@ public class GenericPlayer extends Player {
 	public void sendPacket(SWGPacket packet) {
 		packetLock.lock();
 		try {
+			encodeAndDecode(packet);
 			this.packets.add(packet);
 			packetLockCondition.signalAll();
 		} finally {
@@ -89,7 +91,16 @@ public class GenericPlayer extends Player {
 		}
 		handlePacket(packet);
 	}
-	
+
+	private static void encodeAndDecode(SWGPacket packet) {
+		try {
+			// This doesn't actually test the encoding and decoding of the packet, but it does test that the packet can be encoded and decoded without throwing an exception
+			NetworkProtocol.decode(NetworkProtocol.encode(packet));
+		} catch (Throwable t) {
+			throw new RuntimeException("Failed to encode and decode packet", t);
+		}
+	}
+
 	@Override
 	public void sendPacket(SWGPacket packet1, SWGPacket packet2) {
 		sendPacket(packet1);

--- a/src/test/java/com/projectswg/holocore/test/resources/GenericPlayer.java
+++ b/src/test/java/com/projectswg/holocore/test/resources/GenericPlayer.java
@@ -1,11 +1,10 @@
 /***********************************************************************************
  * Copyright (c) 2024 /// Project SWG /// www.projectswg.com                       *
  *                                                                                 *
- * ProjectSWG is the first NGE emulator for Star Wars Galaxies founded on          *
+ * ProjectSWG is an emulation project for Star Wars Galaxies founded on            *
  * July 7th, 2011 after SOE announced the official shutdown of Star Wars Galaxies. *
- * Our goal is to create an emulator which will provide a server for players to    *
- * continue playing a game similar to the one they used to play. We are basing     *
- * it on the final publish of the game prior to end-game events.                   *
+ * Our goal is to create one or more emulators which will provide servers for      *
+ * players to continue playing a game similar to the one they used to play.        *
  *                                                                                 *
  * This file is part of Holocore.                                                  *
  *                                                                                 *
@@ -30,6 +29,7 @@ package com.projectswg.holocore.test.resources;
 import com.projectswg.common.data.CRC;
 import com.projectswg.common.data.encodables.tangible.Posture;
 import com.projectswg.common.data.location.Location;
+import com.projectswg.common.network.NetBuffer;
 import com.projectswg.common.network.NetworkProtocol;
 import com.projectswg.common.network.packets.SWGPacket;
 import com.projectswg.common.network.packets.swg.zone.*;
@@ -95,7 +95,12 @@ public class GenericPlayer extends Player {
 	private static void encodeAndDecode(SWGPacket packet) {
 		try {
 			// This doesn't actually test the encoding and decoding of the packet, but it does test that the packet can be encoded and decoded without throwing an exception
-			NetworkProtocol.decode(NetworkProtocol.encode(packet));
+			NetBuffer data = NetworkProtocol.encode(packet);
+			NetworkProtocol.decode(data);
+			int remaining = data.remaining();
+			if (remaining > 0) {
+				throw new RuntimeException("Encoded packet buffer had more data that wasn't read during decoding. Bytes remaining: " + remaining);
+			}
 		} catch (Throwable t) {
 			throw new RuntimeException("Failed to encode and decode packet", t);
 		}

--- a/src/test/java/com/projectswg/holocore/test/runners/AcceptanceTest.kt
+++ b/src/test/java/com/projectswg/holocore/test/runners/AcceptanceTest.kt
@@ -26,6 +26,7 @@
  ***********************************************************************************/
 package com.projectswg.holocore.test.runners
 
+import com.projectswg.common.data.encodables.galaxy.Galaxy
 import com.projectswg.common.data.location.Location
 import com.projectswg.holocore.headless.MemoryUserDatabase
 import com.projectswg.holocore.resources.support.data.server_info.loader.ServerData
@@ -65,6 +66,7 @@ import com.projectswg.holocore.services.support.objects.radials.RadialService
 import org.junit.jupiter.api.AfterEach
 import org.junit.jupiter.api.BeforeAll
 import org.junit.jupiter.api.BeforeEach
+import java.time.ZoneOffset
 import java.util.*
 
 /**
@@ -82,11 +84,15 @@ abstract class AcceptanceTest : TestRunnerSynchronousIntents() {
 
 	@BeforeEach
 	fun setUpServices() {
+		val galaxy = Galaxy()
+		galaxy.setZoneOffset(ZoneOffset.UTC)
+		val galaxies = setOf(galaxy)
+
 		registerService(ClientAwarenessService())
 		registerService(CharacterLookupService())
 		registerService(SimulatedObjectStorage())
 		registerService(AwarenessService())
-		registerService(LoginService(memoryUserDatabase))
+		registerService(LoginService(galaxies, memoryUserDatabase))
 		registerService(ZoneService())
 		registerService(CommandQueueService(5, DeterministicDie(0), DeterministicDie(0), DeterministicDie(0), skipWarmup = true))
 		registerService(CommandExecutionService())

